nav2_util/src/bt_loop_rate.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2024 Angsa Robotics
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_util/bt_loop_rate.hpp"
+
+namespace nav2_util
+{
+
+LoopRate::LoopRate(const rclcpp::Duration & period, BT::Tree * tree)
+: clock_(std::make_shared<rclcpp::Clock>(RCL_STEADY_TIME)), period_(period),
+  last_interval_(clock_->now()), tree_(tree)
+{}
```

Comment:
I would have liked to inherit from WallRate to reuse some code but its attributes are private so I can't use them anyway